r-package/grf/R/average_late.R

```diff
@@ -0,0 +1,151 @@
+#' Estimate the average (conditional) local average treatment effect using a causal forest.
+#'
+#' Given an outcome Y, treatment W and instrument Z, the (conditional) local
+#' average treatment effect is tau(x) = Cov[Y, Z | X = x] / Cov[W, Z | X = x].
+#' This is the quantity that is estimated with an instrumental forest.
+#' It can be intepreted causally in various ways. Given a homogeneity
+#' assumption, tau(x) is simply the CATE at x. When W is binary
+#' and there are no "defiers", Imbens and Angrist (1994) show that tau(x) can
+#' be interpreted as an average treatment effect on compliers. This function
+#' is about estimating tau = E[tau(X)] which, extending standard nomenclature,
+#' should perhaps be called the Average (Conditional) Local Averate Treatment
+#' Effect (ACLATE).
+#' 
+#' We estimate the ACLATE using a doubly robust estimator. See Chernozhukov
+#' et al. (2016) for a discussion, and Section 5.2 of Athey and Wager (2017)
+#' for an example using forests.
+#'
+#' If clusters are specified for the forest, then each cluster gets equal weight.
+#' For example, if there are 10 clusters with 1 unit each and per-cluster ATE = 1,
+#' and there are 10 clusters with 19 units each and per-cluster ATE = 0, then the
+#' overall ATE is 0.5 (not 0.05).
+#'
+#' @param forest The trained forest.
+#' @param compliance.score An estimate of the causal effect of Z on W, for
```

Comment:
I had to check the EPL paper to remind myself of what a compliance score was. Perhaps include the formula here?

Reply:
Fair point!